apps/services/user-profile/src/app/v2/user-profile.service.ts

```diff
@@ -36,6 +49,175 @@ export class UserProfileService {
+              }
+            }))
+      }
+
+      if (userProfile.mobilePhoneNumber !== '') {
+        isMobilePhoneNumberDefined &&
+          (await this.verificationService
+            .confirmSms(
+              {
+                mobilePhoneNumber: userProfile.mobilePhoneNumber,
+                code: userProfile.mobilePhoneNumberVerificationCode,
+              },
+              nationalId,
+              transaction,
+            )
+            .then((res) => {
+              if (!res.confirmed) {
+                throw new BadRequestException(res.message)
+              }
+            }))
+      }
+
+      const update = {
+        nationalId,
+        ...(isEmailDefined && {
+          email: userProfile.email === '' ? null : userProfile.email,
+          emailVerified: true,
+        }),
+        ...(isMobilePhoneNumberDefined && {
+          mobilePhoneNumberVerified: true,
+          mobilePhoneNumber:
+            userProfile.mobilePhoneNumber === '' ? null : nationalNumber,
+        }),
+        ...(isDefined(userProfile.locale) && {
+          locale: userProfile.locale,
+        }),
+      }
+
+      await this.userProfileModel.upsert(
+        {
+          ...update,
+          lastNudge: new Date(),
+        },
+        { transaction },
+      )
+
+      await this.islykillService.updateIslykillSettings({
+        nationalId,
+        phoneNumber: parsedPhoneNumber,
+        email: userProfile.email,
+      })
+      return update
+    })
+  }
+
+  async createEmailVerification({
+    nationalId,
+    email,
+  }: {
+    nationalId: string
+    email: string
+  }) {
+    if (!isEmail(email)) {
+      throw new BadRequestException('Email is required')
+    }
+    await this.verificationService.createEmailVerification(nationalId, email, 3)
+  }
+
+  async createSmsVerification({
+    nationalId,
+    mobilePhoneNumber,
+  }: {
+    nationalId: string
+    mobilePhoneNumber: string
+  }) {
+    const { nationalNumber } = this.formatPhoneNumber(mobilePhoneNumber)
+
+    await this.verificationService.createSmsVerification(
+      {
+        nationalId,
+        mobilePhoneNumber: nationalNumber,
+      },
+      3,
+    )
+  }
+
+  async confirmNudge(nationalId: string): Promise<void> {
+    await this.userProfileModel.upsert({ nationalId, lastNudge: new Date() })
+  }
+
+  private checkNeedsNudge(lastNudge: Date | null): boolean {
+    if (!lastNudge) {
+      return null
+    }
+
+    const sixMonthsAgoDate = new Date()
+    sixMonthsAgoDate.setMonth(sixMonthsAgoDate.getMonth() - 6)
+
+    return new Date(lastNudge) < sixMonthsAgoDate
+  }
+
+  private formatPhoneNumber(phoneNumber: string): {
+    nationalNumber: string
+    parsedPhoneNumber: string
+  } {
+    if (phoneNumber === '') {
+      return null
+    }
+
+    let nationalNumber = phoneNumber
+    const tempPhoneNumber = parsePhoneNumber(phoneNumber, 'IS')
+    tempPhoneNumber.country === 'IS' &&
+      (nationalNumber = tempPhoneNumber.nationalNumber as string)
```

Comment:
Personally I find the good old if sentence more readable.

```suggestion
    if (tempPhoneNumber.country === 'IS') {
      nationalNumber = tempPhoneNumber.nationalNumber as string
    }
```